app/controllers/changes_controller.rb

```diff
@@ -23,6 +23,34 @@ def resend
     render json: @change.send_notifications
   end
 
+  def html
+    @change_id=params[:change_id]
+    @snapshot = PageSnapshot.find(@change_id)
```

Comment:
I think this is faulty logic that looks like works because there happen to be the same number of changes as snapshots (all belonging to the same page) in your local database. The `change_id` refers to the `changes_table` but you're using to query the `page_snapshots` table. The correct way to get the relevant snapshot would be with `@change.after` (if you've followed my suggestion above, otherwise `Change.find(params[:change_id]).after`).

Reply:
It actually occurs to me now that I think all of this logic should be on the PageSnapshotController (which doesn't exist currently). What you're doing doesn't actually have anything to do with changes, it's all based on the snapshots. I don't think it's absolutely necessary to move it, but that would be the better organization. This maybe some of the source of the confusion over what the `change_id` refers to.

Reply:
Ahhh. I was wondering about the Pages controller, but could create a PageSnapshotController instead. Got it.

Reply:
Yea, minor mistake I made: controller names are plural, by convention, so it's `PageSnapshotsController`.